src/stdlib/parse_etld.rs

```diff
@@ -0,0 +1,239 @@
+use crate::compiler::prelude::*;
+use std::collections::BTreeMap;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseEtld;
+
+impl Function for ParseEtld {
+    fn identifier(&self) -> &'static str {
+        "parse_etld"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "plus_parts",
+                kind: kind::INTEGER,
+                required: false,
+            },
+        ]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "parse etld",
+                source: r#"parse_etld!("vector.dev")"#,
+                result: Ok(indoc! {r#"
+                {
+                    "etld": "dev",
+                    "etld_plus": "dev",
+                    "known_suffix": true
+                }
+            "#}),
+            },
+            Example {
+                title: "parse etld with plus parts",
+                source: r#"parse_etld!("vector.dev", plus_parts: 1)"#,
```

Comment:
What if by mistake we had `plus_parts: 10` here? Can you add a test for this?

Reply:
There is actually a test for that, named `short_plus`, but it uses `plus_parts: 3` when only 2 are available. I will change it to 10 for better visibility and clarity, but this makes me think if behavior in that case is right. It will just silently return whatever is available (so "vector.dev" in the above case). Would it be better to somehow let it be known in the results that it was not able to get all of these parts, since this way it would require some additional checks on the callers end.

Reply:
> but this makes me think if behavior in that case is right.

Some options to consider:
1. return the max available parts
2. return an error

I am leaning towards 1 which I believe is the current behavior but let me know if you have further thoughts.

Reply:
I agree. That should cover the general case and keep things simple.